taca/analysis/analysis.py

```diff
@@ -147,6 +149,80 @@ def transfer_run(run_dir, analysis):
         runObj.transfer_run(os.path.join("nosync",CONFIG['analysis']['status_dir'], 'transfer.tsv'),
                             analysis, mail_recipients) # do not start analsysis automatically if I force the transfer
 
+
+def transfer_runfolder(run_dir, pid):
+    """ Transfer the entire run folder for a specified project and run to uppmax
+    :param: string run_dir: the run to transfer
+    :param: string pid: the project to include in the SampleSheet
+    """
+    original_sample_sheet = os.path.join(run_dir, "SampleSheet.csv")
+    new_sample_sheet = os.path.join(run_dir, pid + "_SampleSheet.txt")
+
+    # Write new sample sheet including only rows for the specified project
+    try:
+        with open(new_sample_sheet, 'w') as nss:
+            nss.write(extract_project_samplesheet(original_sample_sheet, pid))
+    except:
+        logger.warn("Error parsing the samplesheet")
+        return
+
+    # Create a tar archive of the runfolder
+    dir_name = os.path.basename(run_dir)
+    archive = dir_name + ".tar.gz"
+    run_dir_path = os.path.dirname(run_dir)
+
+    try:
+        subprocess.call(["tar", "--exclude", "Demultiplexing*", "--exclude", "demux_*", "--exclude", "rsync*", "--exclude", "*.csv", "-cvzf", archive, "-C", run_dir_path, dir_name])
+    except:
+        logger.warn("Error creating tar archive")
+        return
+
+    # Generate the md5sum
+    md5file = archive + ".md5"
+    try:
+        f = open(md5file, 'w')
+        subprocess.call(["md5sum", archive], stdout=f)
+        f.close()
+    except:
+        logger.warn("Error creating md5 file")
+        return
+
+    # Rsync the files to irma
+    destination = CONFIG["analysis"]["deliver_runfolder"].get("destination")
+    rsync_opts = {"--no-o" : None, "--no-g" : None, "--chmod" : "g+rw"}
+    connection_details = CONFIG["analysis"]["deliver_runfolder"].get("analysis_server")
+    archive_transfer = RsyncAgent(archive, dest_path=destination, remote_host=connection_details["host"], remote_user=connection_details["user"], validate=False, opts=rsync_opts)
+    md5_transfer = RsyncAgent(md5file, dest_path=destination, remote_host=connection_details["host"], remote_user=connection_details["user"], validate=False, opts=rsync_opts)
+
+    try:
+        archive_transfer.transfer()
+        md5_transfer.transfer()
+    except:
+        logger.warn("Error transferring files to Irma")
+        return
+
+    # clean up the generated files
+    try:
+        os.remove(new_sample_sheet)
+        os.remove(archive)
+        os.remove(md5file)
+    except:
+        logger.warn("Was not able to delete all temporary files")
+
+    return
+
+def extract_project_samplesheet(sample_sheet, pid):
+    header_lines = ""
+    project_entries = ""
+    with open(sample_sheet) as f:
+        for line in f:
+            if not line[0].isdigit():  # include the header
```

Comment:
Hi, I guess this line may not work because flowcell ID may be at line [0]? See example below:
```
[Header]
[Data]
FCID,Lane,Sample_ID,Sample_Name,Sample_Ref,index,index2,Description,Control,Recipe,Operator,Sample_Project
H7LWCDRXX,1,Sample_P13503_1001,P13503_1001,Mouse (Mus musculus GRCm38),AGGCTACC,,A__B_19_01,N,2x50,Expert,A__B_19_01
H7LWCDRXX,1,Sample_P13503_1001,P13503_1001,Mouse (Mus musculus GRCm38),CTAGCTGT,,A__B_19_02,N,2x50,Expert,A__B_19_01
```
Otherwise it looks great! Thanks for the work!

Reply:
Ah, shoot. Will have another look at that!